### PR TITLE
US193682: ACS: Refactor integration tests listeners and grouping to remove dependency on cloud foundry packages

### PR DESCRIPTION
**Pull request by VoollaSandeepKumar**
US193682: ACS: Refactor integration tests listeners and grouping to remove dependency on cloud foundry packages

### CODE REVIEW

File: acs-integration-tests/src/test/java/com/ge/predix/test/TestNameLogger.java [hunk 1]
```diff
@@ -22,12 +22,9 @@
 import org.testng.IInvokedMethodListener;
 import org.testng.ITestNGMethod;
 import org.testng.ITestResult;
-import org.testng.SkipException;
-
-import com.ge.predix.acs.cloudfoundry.DeleteApplications;
 
 public class TestNameLogger implements IInvokedMethodListener {
-    private enum TestStatus {
+    public enum TestStatus {
```
Comment (irinaepshteyn) [inline acs-integration-tests/src/test/java/com/ge/predix/test/TestNameLogger.java]: protected or package access level should be enough.

File: acs-integration-tests/src/test/java/com/ge/predix/test/TestNameLogger.java [hunk 2]
```diff
@@ -47,11 +44,8 @@ public String toString() {
     }
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TestNameLogger.class);
-    static final String ACS_CLOUD_FOUNDRY_PACKAGE = "com.ge.predix.acs.cloudfoundry";
-
-    private static volatile boolean suiteFailed = false;
 
-    private static void logInvocation(final TestStatus testStatus, final IInvokedMethod method) {
+    public static void logInvocation(final TestStatus testStatus, final IInvokedMethod method) {
```
Comment (irinaepshteyn) [inline acs-integration-tests/src/test/java/com/ge/predix/test/TestNameLogger.java]: protected or package access level should be enough.

File: acs-integration-tests/src/test/resources/log4j.xml [hunk 3]
```diff
@@ -58,6 +58,10 @@
         <level value="DEBUG"/>
     </logger>
 
+    <logger name="com.ge.predix.test.TestNamePredixLogger">
+        <level value="DEBUG"/>
+    </logger>
+
```
Comment (irinaepshteyn) [inline acs-integration-tests/src/test/resources/log4j.xml:64]: Should we turn on DEBUG for the TestNameLogger base class as well?

File: acs-integration-tests/src/test/java/com/ge/predix/test/TestNamePredixLogger.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    public void beforeInvocation(final IInvokedMethod method, final ITestResult testResult) {
+        // Skip the test if the suite has been marked as failing and the test being executed isn't a
+        // Cloud-Foundry-related deletion operation (i.e. a normal integration test)
+        if (skipTest(method)) {
+            throw new SkipException("Test skipped due to a failure detected in the suite");
+        }
+
+        logInvocation(TestStatus.STARTING, method);
```
Comment (irinaepshteyn) [inline acs-integration-tests/src/test/java/com/ge/predix/test/TestNamePredixLogger.java]: We could call super here to do the logging.

File: acs-integration-tests/src/test/java/com/ge/predix/test/TestNamePredixLogger.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            return;
+        }
+
+        if (skipTest(method)) {
+            logInvocation(TestStatus.SKIPPING, method);
+            return;
+        }
+
+        logInvocation((testResult.getStatus() == ITestResult.FAILURE ? TestStatus.ERRORED_OUT : TestStatus.FINISHING),
+                method);
```
Comment (irinaepshteyn) [inline acs-integration-tests/src/test/java/com/ge/predix/test/TestNamePredixLogger.java]: We could call super here to do the logging.

File: acs-integration-tests/src/test/java/com/ge/predix/test/TestNamePredixLogger.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package com.ge.predix.test;
```
Comment (irinaepshteyn) [inline acs-integration-tests/src/test/java/com/ge/predix/test/TestNamePredixLogger.java:17]: Add licensing header.

**Review by irinaepshteyn — CHANGES_REQUESTED** [review]
Done reviewing.  Please address my comments.

File: acs-integration-tests/src/test/java/com/ge/predix/test/TestNameLogger.java (hunk 1 above)
Comment (VoollaSandeepKumar) [reply acs-integration-tests/src/test/java/com/ge/predix/test/TestNameLogger.java]: Changed to Protected

File: acs-integration-tests/src/test/java/com/ge/predix/test/TestNameLogger.java (hunk 2 above)
Comment (VoollaSandeepKumar) [reply acs-integration-tests/src/test/java/com/ge/predix/test/TestNameLogger.java]: Changed to Protected

File: acs-integration-tests/src/test/java/com/ge/predix/test/TestNamePredixLogger.java (hunk 6 above)
Comment (VoollaSandeepKumar) [reply acs-integration-tests/src/test/java/com/ge/predix/test/TestNamePredixLogger.java:17]: Fixed

File: acs-integration-tests/src/test/resources/log4j.xml (hunk 3 above)
Comment (VoollaSandeepKumar) [reply acs-integration-tests/src/test/resources/log4j.xml:64]: I think its already added

**Review by VoollaSandeepKumar — COMMENTED** [review]
Fixed

**Review by irinaepshteyn — APPROVED** [review]
Approve.  Please squash commits and merge.

**Pull request merged**